### PR TITLE
feat: add Odin

### PR DESCRIPTION
**Pull request by amaanq**
cc: @laytan @WillEhrendreich

@laytan Give this a shot :grin:

### CODE REVIEW

File: queries/odin/indents.scm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+] @indent.begin
+
+[
+  "}"
+  ")"
+  "]"
+] @indent.branch @indent.end
+
+[
+  (comment)
```
Comment (lucario387) [inline queries/odin/indents.scm:23]: `(block_comment)`

**Comment by laytan** [comment]
This looks great and is a better implementation than I hacked together.

One thing I would like to see is a highlight for an explicit enum: `.Foo` or `.Test` (basically a member expression with only one identifier if I look at your grammar, would probably be an @ type, not sure if there is something for enums.

Another thing, make 'context' a @ builtin.variable.

I will close my PR

**Comment by laytan** [comment]
One more thing I noticed is that this gets highlighted as a function, but it is actually a type (because it has no body):
`Handler_Proc :: proc(next: ^Handler, req: ^Request, res: ^Response)`

**Comment by amaanq** [comment]
@laytan I fixed the freeze with regards to brackets, can you try it out a bit more and let me know of other potential bugs?

Also the proc types vs functions & context builtin issues are fixed, but can you elaborate more on the explicit enums?

**Comment by laytan** [comment]
@amaanq In this snippet for example:
```odin
method_string :: proc(m: Method) -> string {
	switch m {
	case .Get:     return "GET"
	case .Head:    return "HEAD"
	case .Post:    return "POST"
	case .Put:     return "PUT"
	case .Patch:   return "PATCH"
	case .Trace:   return "TRACE"
	case .Delete:  return "DELETE"
	case .Connect: return "CONNECT"
	case .Options: return "OPTIONS"
	case:          return ""
	}
}
```

The compiler knows here that we are working with the Status enum, in the cases where it does, you can leave out the `Status` in `Status.Get`, so `Status.Get` == `.Get`. This is only possible with enums, so you can highlight that as an enum/type.

Here is the official overview's section about them: http://odin-lang.org/docs/overview/#implicit-selector-expression

Your other fixes and addition are working great btw!

**Comment by amaanq** [comment]
so...would `@constant` be appropriate?

**Comment by laytan** [comment]
@amaanq Its an enum, I have no clue what other languages with enums highlight that as, I think you know that better than I do ;)

**Comment by amaanq** [comment]
yeah constant, regular member expressions (foo.bar) are fields, should be good now then.

File: queries/odin/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+((ternary_expression
+  [
+    "?"
+    ":"
+    "if"
+    "else"
+    "when"
+  ] @conditional.ternary)
+  (#set! "priority" 105))
```
Comment (lucario387) [inline queries/odin/highlights.scm:67]: Hm, can't this be done in a way that doesn't require tweaking priority?

Comment (amaanq) [reply queries/odin/highlights.scm:67]: It's because of ":" in punctuation as well, I could move it below punctuation, but I like where it is now

**Review by lucario387 — APPROVED** [review]

**Pull request merged**